### PR TITLE
feat: Add a maximum Session lifespan

### LINKED ISSUE #231: Introduce max session length
**Problem:**
Today, we don't have a maximum user session length. This can result in very long sessions for over an hour that would...
- Require a lot of storage
- Result in slow performance on Replay views in the Sentry app because of data volume 
- Make it more likely we drop segments and have low data quality

Datadog has a maximum user session length of 30 minutes. 

**Possible Solution:**

Set maximum session length to be 30 minutes.

[**Side note:** _We should also have controls to trim the session length to a user-defined max, as long as it doesn't go over the max we have set in the SDK_.]

### PR DESCRIPTION
**Pull request by billyvg**
... of 30 minutes.

Fixes #231

### CODE REVIEW

File: src/util/isSessionExpired.ts [hunk 1]
```diff
@@ -10,5 +11,8 @@ export function isSessionExpired(
   expiry: number,
   targetTime = +new Date()
 ) {
-  return isExpired(session?.lastActivity, expiry, targetTime);
+  return (
+    isExpired(session.started, MAX_SESSION_LIFE, targetTime) ||
+    isExpired(session?.lastActivity, expiry, targetTime)
```
Comment (JoshFerge) [inline src/util/isSessionExpired.ts:16]: could we rename the `expiry` variable here to `idle_expiry` or something? confusing for me on first read

**Review by JoshFerge — APPROVED** [review]

File: src/util/isSessionExpired.ts (hunk 1 above)
Comment (billyvg) [reply src/util/isSessionExpired.ts:16]: hmm, what about `expiryLimit`?

Comment (JoshFerge) [reply src/util/isSessionExpired.ts:16]: `expiryLimit` still doesn't really tell me what that means, have to go all the way to index.ts to understand what the context of `expiry` is -- is that value only used for idle timeouts?

Comment (billyvg) [reply src/util/isSessionExpired.ts:16]: Yeah I think at one point it was possible to expire for non-idle reasons, but not anymore

**Pull request merged**